renku_notebooks/api/notebooks.py

```diff
@@ -121,16 +120,47 @@ def launch_notebook(user):
     for key in server_options_defaults.keys():
         server_options.setdefault(key, server_options_defaults.get(key)["default"])
 
-    gl_project = get_project(user, namespace, project)
+    gl_project = get_renku_project(user, f"{namespace}/{project}")
 
     if gl_project is None:
         return current_app.response_class(
             status=404,
             response=f"Cannot find project {project} for user: {user['name']}.",
         )
 
-    # set the notebook image
-    image = get_notebook_image(user, namespace, project, commit_sha)
+    # set the notebook image if one is not
+    if requested_image is None:
+        parsed_image = {
+            "hostname": config.IMAGE_REGISTRY,
+            "image": gl_project.path_with_namespace.lower(),
+            "tag": commit_sha[:7],
+        }
+        commit_image = (
+            f"{config.IMAGE_REGISTRY}/{gl_project.path_with_namespace.lower()}"
+            f":{commit_sha[:7]}"
+        )
+    else:
+        parsed_image = parse_image_name(requested_image)
+    # get token
+    token, is_image_private = get_docker_token(**parsed_image, user=user)
+    # check if images exist
+    image_exists_result = image_exists(**parsed_image, token=token)
+    # assign image
+    if image_exists_result and requested_image is None:
+        # the image tied to the commit exists
+        image = commit_image
+    elif not image_exists_result and requested_image is None:
+        # the image tied to the commit does not exist, fallback to default image
+        image = config.DEFAULT_IMAGE
+        is_image_private = False
+    elif image_exists_result and requested_image is not None:
+        # a specific image was requested and it exists
+        image = requested_image
+    else:
+        # a specific image was requested but does not exist
+        return make_response(
+            jsonify({"error": f"Cannot find image {requested_image}."}), 404
```

Comment:
```suggestion
            jsonify({"error": f"Cannot find/access image {requested_image}."}), 404
```
Just a little hint that it might be an auth problem as well.